pageserver/src/tenant/remote_timeline_client/upload.rs

```diff
@@ -114,16 +113,16 @@ pub(crate) async fn upload_initdb_dir(
     storage: &GenericRemoteStorage,
     tenant_id: &TenantId,
     timeline_id: &TimelineId,
-    initdb_dir: Bytes,
+    initdb_tar_zst: File,
+    size: u64,
 ) -> anyhow::Result<()> {
     tracing::trace!("uploading initdb dir");
 
-    let size = initdb_dir.len();
-    let bytes = tokio::io::BufReader::new(std::io::Cursor::new(initdb_dir));
+    let bytes = tokio::io::BufReader::new(initdb_tar_zst);
 
     let remote_path = remote_initdb_archive_path(tenant_id, timeline_id);
     storage
-        .upload_storage_object(bytes, size, &remote_path)
+        .upload_storage_object(bytes, size as usize, &remote_path)
```

Comment:
I will conflict this with #5446.

Reply:
I have merged latest main now and resolved the conflicts